### DISCUSSION TITLE
How to make a conditional statement?

### DISCUSSION THREAD

**Discussion by DBonbon** [vercel/next.js#50435] [Help] [answered]
### Summary

I try to iterate a block with 4 items and iterate:
the item imagethe items title 

The titles of the other 3 items.
Assuming that the image and the item.title are extracted correctly. How can I extract the other 3 items' title?ex: card.value.cardTitle
I googled and tried the following - i.e. compare the index against  a loop 'i' variable so when index===i it returns the imageTitle in red, else it returns a blue imgaeTitle. 

Any input, and/or reference to relevant sources would be highly appreciated. 

```class CardsPage extends PureComponent {
    state = {};
 ....
    

render() {
     const { title, comments, cards } = this.props;
     return (
         <Card>
         <Card.Header>
             <Col>
             <p>serie title: {title}</p>
             <RawHtml html={comments} />
             </ Col>
             </ Card.Header>
              <div>
     {cards.map((card, index) => (
     <Card.Body>
         <Text>
         {card.value.image.title}
         </Text>
         <Card.Image
         src={card.value.image.original.src}
         alt="BigCo Inc. logo"/>
         
         <p style={{ color : "red" }}> 
          {card.value.cardTitle}</p>
         
         for( let i=0; i<index; i++) {
           if( i === index ) {
           <p> {card.value.cardTitle}</p>} 
           else {
           <p style={{ color : "blue" }}>
           {card.value.cardTitle}</p>} 

           </p>
           i+=1
           }
          }
    </ Card.Body>
    ))}

        );
    }
}

export default basePageWrap(CardsPage);```

**Comment by devjiwonchoi** [comment]
I do not get the point where you want to render one title red and the other blue.  
Could you elaborate on the precise condition for that?

For example, do you want to render red, then three blues, then red?
Such as red, blue, blue, blue, red, blue...etc?

**Reply by DBonbon** [reply]
Sorry for the lack of clarity. 
I wish to iterated through a block with four items. Each item has a title and an image. 
I try to render them like a quartet card game.
I wish it renders the image and title as well as the 3 other titles. 

Say I have 4 values: shark, whale, dolphin, seal

Instead of extracting image: 
I wish to render
<image-of a given iterated-sea mammal>
**<title-of a given iterated-sea mammal>*
and the three title that are not in the given iteration. 

Ex:
Whale Image

1. Shark 
2. **Whale** (bold for the iterated 'red' item)
3. Dolphin
4. Seal

**Reply by devjiwonchoi** [reply]
First of all, as @icyJoseph stated below, in `JSX` format you cannot use `if/else` or `for` inside the `return` just like a regular Javascript function.
For more information, look [here](https://nextjs.org/learn/foundations/from-javascript-to-react/displaying-data-with-props) and [here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/return#see_also).

In your case, you can use `map` and `filter` since they **return a new array**.

For example,

```
{props.map((mainCard) => {
  <CardBody>
    <Text>{mainCard.title}</Text>
    <CardImage src={mainCard.imageSrc} alt='BigCo Inc. logo' />
  
    <p style={{ color: 'red' }}>{mainCard.cardTitle}</p>
  
    {props
      .filter((el) => el.title !== mainCard.title)
      .map((subCard) => {
        <p style={{ color: 'blue' }}>{subCard.cardTitle}</p>;
      })}
  </CardBody>;
})}
```

**Comment by icyJoseph** [comment, marked as answer]
In JSX you cannot inline if/else, while nor for loops just like that. 

In your case there's various ways to fix it, for example:

```jsx
 const { title, comments, cards } = this.props;
  return (
    <Card>
      <Card.Header>
        <Col>
          <p>serie title: {title}</p>
          <RawHtml html={comments} />
        </Col>
      </Card.Header>
      <div>
        {/* the third argument of the map callback is the list itself*/}
        {cards.map((card, _, src) => (
          <Card.Body key={card.id}>
            <Text>{card.value.image.title}</Text>
            <Card.Image
              src={card.value.image.original.src}
              alt="BigCo Inc. logo"
            />

            {/* do you really want this one? */}
            <p style={{ color: "red" }}>{card.value.cardTitle}</p>

            {src.map((nestedCard) => {
              // compare by reference, do we have the same top card down here?
              if (nestedCard === card) {
                // maybe this one ought to be red?
                return <p>{nestedCard.value.cardTitle}</p>;
              }
              return <p style={{ color: "blue" }}>{nestedCard.value.cardTitle}</p>;
            })}
          </Card.Body>
        ))}
      </div>
    </Card>
  );
```

I think it is a good idea to practice different syntax patterns and such, in isolation, before getting swamped by complexity.

Talking about complexity, for every item in the list, here you walk the list again, making this O(n^2) right? If you have 4 items, you walk the list 16 times in total.

One important bit, I added a key prop to the `Card` element. The usage of `key` is vital when rendering lists! I assumed the card has an `id`, but if not, then you just need something locally unique, preferrably not the index (position) in the list.

Other bits:

- I used the 3rd argument of the map function, which is the source list itself.
- I skipped the index by placing a `_` but that's just a convention, any name works.
- Different syntax for arrow functions return.
- Comparison by reference `nestedCard === card`

Further:

```jsx
  {src.map((nestedCard) => {
              // compare by reference, do we have the same top card down here?
              const style =
                nestedCard === card ? { color: "red" } : { color: "blue" };

              return <p style={style}>{nestedCard.value.cardTitle}</p>;
            })}
```

You can also just use a ternary to calculate the style object applied to each `p` tag :) - and there's much more...

**Reply by icyJoseph** [reply]
Also, technically this didn't answer the question, but your intention with the code.

To do a conditional you do:

```jsx
{Condition ? <div /> : <span />}
```

- Docs: https://react.dev/learn/conditional-rendering

**Reply by icyJoseph** [reply]
@DBonbon did you manage to resolve this?

**Reply by DBonbon** [reply]
Hi, @icyJoseph a. Yes I did... thanks u, thanks u  b. Sorry, sorry for my belated reply. I actually solved a moment after I got your reply, but forgot to note it. My bad.
